### PR TITLE
[Java] Simplify ray cmd params

### LINKED ISSUE #10432: Simplify options users need to specify when starting a Java driver and connecting to an existing Ray cluster
Below options are unnecessary and hard to be understood by users:

* `ray.object-store.socket-name`
* `ray.raylet.socket-name`
* `ray.raylet.node-manager-port`
* `ray.node-ip`

### PR DESCRIPTION
**Pull request by chaokunyang**
## Why are these changes needed?
Remove nodeIp/objectStoreSocketName/rayletSocketName/nodeManagerPort from driver command line

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java [hunk 1]
```diff
@@ -80,6 +83,40 @@
...
+      // Keep this method logic in sync with `services.get_address_info_from_redis_helper`
+      int numRetries = 5;
+      int retryCount = 0;
+      boolean loadConfig = false;
+      while (retryCount++ < numRetries) {
+        for (NodeInfo nodeInfo : tempGcsClient.getAllNodeInfo()) {
+          if (rayConfig.nodeIp.equals(nodeInfo.nodeAddress) ||
+              (nodeInfo.nodeAddress.equals("127.0.0.1") &&
+                  rayConfig.nodeIp.equals(rayConfig.getRedisAddress()))) {
+            rayConfig.objectStoreSocketName = nodeInfo.objectStoreSocketName;
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:96]: This whole section should be put into the constructor instead of the static block. The `rayConfig` here is just a local variable. I don't understand how it works because the changes you made here will be lost.

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java [hunk 2]
```diff
@@ -80,6 +83,40 @@
           tempGcsClient.getInternalConfig().entrySet()) {
         rayConfig.rayletConfigParameters.put(entry.getKey(), entry.getValue());
       }
+
+      // Keep this method logic in sync with `services.get_address_info_from_redis_helper`
+      int numRetries = 5;
+      int retryCount = 0;
+      boolean loadConfig = false;
+      while (retryCount++ < numRetries) {
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:91]: We should add a worker type check before reading from GCS. This is only needed for driver process.

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java [hunk 3]
```diff
@@ -117,6 +118,7 @@ public RayConfig(Config config) {
     WorkerType localWorkerMode;
     try {
       localWorkerMode = config.getEnum(WorkerType.class, "ray.worker.mode");
+      System.out.println("localWorkerMode" + localWorkerMode);
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:121]: Remove debug code

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java [hunk 4]
```diff
@@ -130,6 +132,28 @@ public RayConfig(Config config) {
       nodeIp = NetworkUtil.getIpAddress(null);
     }
     this.nodeIp = nodeIp;
+    // Redis configurations.
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:135]: Why do you move this block of code upward?

File: python/ray/services.py [hunk 5]
```diff
@@ -71,11 +71,6 @@
     os.path.abspath(os.path.dirname(__file__)),
     "core/src/ray/cpp/default_worker" + EXE_SUFFIX)
 
-DEFAULT_JAVA_WORKER_CLASSPATH = [
```
Comment (kfstorm) [inline python/ray/services.py:74]: I assume this change is unrelated.

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java (hunk 1 above)
Comment (chaokunyang) [reply java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:96]: `RayConfig` is a singleton. We need to use `ray config` to load native binary and library to specified directory. If we do this in the `RayConfig` constrcutor, how can we get ray config instance?

Comment (kfstorm) [reply java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:96]: There's actually a `reset()` method in `RayConfig`, so `RayConfig` is not fully singleton. The global instance will be destroyed during the shutdown process. I was saying we should put this into the constructor of `RayNativeRuntime`, not of `RayConfig`.

Comment (chaokunyang) [reply java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:96]: Fixed

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java (hunk 4 above)
Comment (chaokunyang) [reply java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:135]: reverted

File: python/ray/services.py (hunk 5 above)
Comment (chaokunyang) [reply python/ray/services.py:74]: Will remove it in another PR

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java [hunk 6]
```diff
@@ -91,6 +85,50 @@ private static void resetLibraryPath(RayConfig rayConfig) {
 
   public RayNativeRuntime(RayConfig rayConfig) {
     super(rayConfig);
+    if (rayConfig.getRedisAddress() != null) {
+      GcsClient tempGcsClient =
```
Comment (raulchen) [inline java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:89]: Please put this in a separate method.

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java [hunk 7]
```diff
@@ -91,6 +85,50 @@ private static void resetLibraryPath(RayConfig rayConfig) {
...
+      for (Map.Entry<String, String> entry :
+          tempGcsClient.getInternalConfig().entrySet()) {
+        rayConfig.rayletConfigParameters.put(entry.getKey(), entry.getValue());
+      }
+
+      if (rayConfig.workerMode == WorkerType.DRIVER) {
+        // Keep this method logic in sync with `services.get_address_info_from_redis_helper`
+        int numRetries = 5;
+        int retryCount = 0;
+        boolean loadConfig = false;
```
Comment (raulchen) [inline java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:100]:
```suggestion
        boolean configLoaded = false;
```

**Review by raulchen — COMMENTED** [review]
Should we also remove the config items that are no longer needed from `ray.default.conf`?

**Comment by chaokunyang** [comment]
@raulchen Removed node-ip/socket-name from `ray.default.conf`

**Review by raulchen — APPROVED** [review]

**Review by kfstorm — APPROVED** [review]
LGTM. Let's merge the lastest master and see if it breaks CI.
